pkg/network/proxy/unidler/unidlerproxy.go

```diff
@@ -35,14 +35,14 @@ func (sig *eventSignaler) NeedPods(serviceName types.NamespacedName, port string
 	}
 
 	// HACK: make the message different to prevent event aggregation
-	sig.recorder.Eventf(&serviceRef, v1.EventTypeNormal, unidlingapi.NeedPodsReason, "The service-port %s:%s needs pods.", serviceRef.Name, port)
+	sig.recorder.Eventf(&serviceRef, nil, v1.EventTypeNormal, unidlingapi.NeedPodsReason, "None", "The service-port %s:%s needs pods.", serviceRef.Name, port)
```

Comment:
```suggestion
	sig.recorder.Eventf(&serviceRef, nil, v1.EventTypeNormal, unidlingapi.NeedPodsReason, "Unidling", "The service-port %s:%s needs pods.", serviceRef.Name, port)
```

Reply:
I disagree with this `action`. SDN is not performing this action - another software component is performing this action and SDN is requesting it via this event. Let me know what you think.

From client-go event recorder interface [1]:
> 	// 'action' explains what happened with regarding/what action did the **ReportingController**
	// (ReportingController is a type of a Controller reporting an Event, e.g. k8s.io/node-controller, k8s.io/kubelet.)
	// take in regarding's name; it should be in UpperCamelCase format (starting with a capital letter).
	// 'note' is intended to be human readable.

[1] https://github.com/kubernetes/client-go/blob/9b0b23a8ade2b5323d6624146cea2ad7b8928f25/tools/events/interfaces.go#L42

Reply:
I have not seen any precedent in upstream k8 code using this `None` action label.

Reply:
Yeah, I read that as, there are two ways you can use `action`; it either "explains what happened with regarding" or "explains what action did the ReportingController take in regarding's name". `"Unidling"` fits the former case.